### ISSUE TITLE
HMC adaptation stuck at warmup phase due to step_size -> nan

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#249]
A script to replicate this issue
```
import jax.numpy as np
from jax.scipy.special import logsumexp
from numpyro.mcmc import mcmc

def dual_moon_pe(x):
    term1 = 0.5 * ((np.linalg.norm(x, axis=-1) - 2) / 0.4) ** 2
    term2 = -0.5 * ((x[..., :1] + np.array([-2., 2.])) / 0.6) ** 2
    return term1 - logsumexp(term2, axis=-1)

mcmc(1000, 1000, init_params=np.array([2., 0.]), potential_fn=dual_moon_pe)
```

**Comment by neerajprad** [comment]
This seems to run fine on my system, even for much higher warmup steps, but gives poor results, as you might expect.
